### PR TITLE
Enhance `SkinnedMeshRenderer` bounds and fix `MeshRenderer` bounds update bug

### PR DESCRIPTION
**Pull request by GuoLei1990**
### Please check if the PR fulfills these requirements

- [x] The commit message follows our [guidelines](https://github.com/oasis-engine/engine/blob/main/.github/COMMIT_MESSAGE_CONVENTION.md)
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

### What kind of change does this PR introduce? 
**SkinnedMeshRenderer**
- Fix `SkinnedMeshRenderer` world bounds update bug
- Add the `localbounds` property for `SkinnedMeshRenderer`, allowing developers to manually adapt in extreme cases
- Optimization code and performance with custom matrix multiply

**MeshRenderer**
- Fix `MeshRenderer` world bouds not update bug when local bounds modified
- Fix `MeshRenderer` world bouds not update bug when mesh modified

**Optimization**

- Logic related to bounds
- Logic related to dirty flag

In addition, `SkinnedMeshRenderer` design is not very good, this PR is just fix bounds problem,It will be optimized later！

### What is the current `SkinnedMeshRenderer` bounds behavior?
- The boudns is static and often exceed boundaries when playing animation.
- Export gltf with params `keep current frame` in blender will cause the bounds very small( bind pose matrix cause)

#### Examples(Wire frame is AABB boudding box)
Rumba Dancing(T-pose)

Falling(T-pose)

Rumba Dancing(keep current frame)

### What is the `SkinnedMeshRenderer` bounds new behavior?

- The bounds can update with the root bone.
-  fix probm with export gltf with params `keep current frame` in blender.

#### Examples(Wire frame is AABB boudding box)
Rumba Dancing(T-pose)

Falling(T-pose)

Rumba Dancing(keep current frame)

### GLTF files
[RumbaDancing.gltf.zip](https://github.com/oasis-engine/engine/files/9878866/RumbaDancing.gltf.zip)

[Falling.gltf.zip](https://github.com/oasis-engine/engine/files/9878867/Falling.gltf.zip)

[RumbaDance(keep-current-frame).gltf.zip](https://github.com/oasis-engine/engine/files/9878869/RumbaDance.keep-current-frame.gltf.zip)

### CODE REVIEW

File: packages/core/src/2d/sprite/SpriteMask.ts [hunk 1]
```diff
@@ -185,10 +168,7 @@ export class SpriteMask extends Renderer implements ICustomClone {
   _onDestroy(): void {
     this._sprite = null;
     this._renderData = null;
-    if (this._spriteChangeFlag) {
-      this._spriteChangeFlag.destroy();
-      this._spriteChangeFlag = null;
-    }
+    this._sprite?._updateFlagManager.removeListener(this._onSpriteChange);
```
Comment (cptbtptpbcptdtptp) [inline packages/core/src/2d/sprite/SpriteMask.ts:171]: Should not be after `this._sprite = null; ` here.

File: packages/core/src/UpdateFlagManager.ts [hunk 2]
```diff
@@ -26,13 +29,45 @@ export class UpdateFlagManager {
...
+    if (success) {
+      removeFromArray(flag._flagManagers, this);
+    }
+  }
+
+  /**
+   * Add a listener.
+   * @param listener - The listener
+   */
+  addListener(listener: (bit?: number, param?: Object) => void): void {
```
[the remark this answers is not in the document]
Comment (GuoLei1990) [reply packages/core/src/UpdateFlagManager.ts]: done

File: packages/core/src/2d/sprite/SpriteMask.ts (hunk 1 above)
Comment (GuoLei1990) [reply packages/core/src/2d/sprite/SpriteMask.ts:171]: done

File: packages/loader/src/gltf/parser/SkinParser.ts [hunk 3]
```diff
@@ -39,12 +40,48 @@ export class SkinParser extends Parser {
...
   }
+
+  private _findSkeletonRootBone(joints: number[], entities: Entity[]): Entity {
+    const paths = <Record<number, Entity[]>>{};
+    for (const index of joints) {
+      const path = new Array<Entity>();
+      let entity = entities[index];
+      while (entity.parent !== null) {
+        path.unshift(entity);
+        entity = entity.parent!;
```
Comment (zhuxudong) [inline packages/loader/src/gltf/parser/SkinParser.ts]: !

Comment (GuoLei1990) [reply packages/loader/src/gltf/parser/SkinParser.ts]: done

File: packages/core/src/mesh/SkinnedMeshRenderer.ts [hunk 4]
```diff
@@ -153,40 +226,68 @@ export class SkinnedMeshRenderer extends MeshRenderer {
...
+    const jointEntitys = new Array<Entity>(jointCount);
+    for (let i = jointCount - 1; i >= 0; i--) {
+      jointEntitys[i] = this._findByEntityName(this.entity, joints[i]);
+    }
+    this._jointEntitys = jointEntitys;
+    this._jointMatrixs = new Float32Array(jointCount * 16);
+
+    const lastRootBone = this._rootBone;
+    const rootBone = this._findByEntityName(this.entity, skin.skeleton);
+    const rootInddex = joints.indexOf(skin.skeleton);
```
[the remark this answers is not in the document]
Comment (GuoLei1990) [reply packages/core/src/mesh/SkinnedMeshRenderer.ts]: done

File: packages/core/src/2d/sprite/SpriteMask.ts [hunk 5]
```diff
@@ -83,8 +79,9 @@ export class SpriteMask extends Renderer implements ICustomClone {
   set height(value: number) {
     if (this._height !== value) {
       this._height = value;
-      this._dirtyFlag |= DirtyFlag.Position;
+      this._dirtyUpdateFlag |= RendererUpdateFlags.WorldVolume;
     }
+    Promise
```
[the remark this answers is not in the document]
Comment (GuoLei1990) [reply packages/core/src/2d/sprite/SpriteMask.ts]: done

**Review by cptbtptpbcptdtptp — APPROVED** [review]
+1

**Pull request merged**